packages/devtools_app/lib/src/screens/inspector_v2/layout_explorer/box/box.dart

```diff
@@ -441,3 +454,95 @@ class BoxChildVisualizer extends StatelessWidget {
     );
   }
 }
+
+/// Encapsulation of [widths] and [heights] for the layout.
+class LayoutWidthsAndHeights {
+  LayoutWidthsAndHeights({
+    required this.widths,
+    required this.heights,
+    required this.availableWidth,
+    required this.availableHeight,
+  })  : assert(widths.length == 3),
+        assert(heights.length == 3) {
+    _displayWidths = _simpleFractionalLayout(
+      availableSize: availableWidth,
+      sizes: widths,
+    );
+    _displayHeights = _simpleFractionalLayout(
+      availableSize: availableHeight,
+      sizes: heights,
+    );
+  }
+
+  final List<double> widths;
+  final List<double> heights;
+  late List<double> _displayWidths;
+  late List<double> _displayHeights;
+  final double availableWidth;
+  final double availableHeight;
+
+  double get widgetWidth => widths[1];
+
+  double get widgetHeight => heights[1];
+
+  double get leftPadding => widths[0];
+
+  double get rightPadding => widths[2];
+
+  double get topPadding => heights[0];
+
+  double get bottomPadding => heights[2];
+
+  double get displayWidgetWidth => _displayWidths[1];
+
+  double get displayWidgetHeight => _displayHeights[1];
+
+  double get displayLeftPadding => _displayWidths[0];
+
+  double get displayRightPadding => _displayWidths[2];
+
+  double get displayTopPadding => _displayHeights[0];
+
+  double get displayBottomPadding => _displayHeights[2];
+
+  bool get hasLeftPadding => leftPadding != 0;
+
+  bool get hasRightPadding => rightPadding != 0;
+
+  bool get hasTopPadding => topPadding != 0;
+
+  bool get hasBottomPadding => bottomPadding != 0;
+
+  bool get hasAnyPadding =>
+      hasLeftPadding || hasRightPadding || hasTopPadding || hasBottomPadding;
+
+  /// Simplistic layout algorithm that will return sizes for the padding, width,
+  /// and height of the widget based on the display's [availableSize] along with
+  /// [sizes] of the widget.
+  ///
+  /// Uses a constant [paddingFraction] for the display padding, regardless of
+  /// the actual size.
+  ///
+  /// The return value and [sizes] parameter correspond to either:
+  /// - left padding width, widget width, right padding width
+  /// - top padding height, widget height, bottom padding height
+  List<double> _simpleFractionalLayout({
+    required double availableSize,
+    required List<double?> sizes,
+  }) {
+    final paddingASize = sizes[0] ?? 0;
+    final paddingBSize = sizes[2] ?? 0;
+
+    final paddingFraction = paddingASize != 0 && paddingBSize != 0 ? 0.3 : 0.35;
+
+    final paddingAFraction = paddingASize > 0 ? paddingFraction : 0.0;
+    final paddingBFraction = paddingBSize > 0 ? paddingFraction : 0.0;
```

Comment:
how about `max(0.0, paddingASize)` and `max(0.0, paddingBSize)` instead?

Reply:
We need to return `paddingFraction` here instead of the padding size.